### ISSUE TITLE
Migration failed, seems can't disable ehcache

### ISSUE THREAD

**Issue by YUFAN2GA** [KarelCemus/play-redis#153]
1. Version of play-redis
2.0.1
1. Version of Scala
2.12.4
1. Version of Play
2.6.10
1. Describe the environment if applies (usually in cloud etc.)
win10 laptop
1. Describe the expected behavior
I used to use 1.4.2 play-redis and play 2.5.4 with scala 2.11.7, it works good, and then I want to upgrade play to 2.6.10 and sbt 1.x and scala 2.12, and follow the migration guide, it should still work

1. Describe the actual behavior, follow the same scenario
following errors were thrown:

> 2) A binding to play.api.cache.AsyncCacheApi was already configured at play.api.cache.ehcache.EhCacheModule$$anonfun$$lessinit$greater$1.bindDefault$1(EhCacheApi.scala:85):
> Binding(interface play.api.cache.AsyncCacheApi to BindingKeyTarget(interface play.api.cache.AsyncCacheApi qualified with QualifierInstance(@play.cache.NamedCache(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1).
>   at play.api.cache.redis.GuiceProvider$.defaultBinding$1(RedisCacheModule.scala:73):
> Binding(interface play.api.cache.AsyncCacheApi to BindingKeyTarget(interface play.api.cache.AsyncCacheApi qualified with QualifierInstance(@javax.inject.Named(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1)
> 
> 3) A binding to play.api.cache.SyncCacheApi was already configured at play.api.cache.ehcache.EhCacheModule$$anonfun$$lessinit$greater$1.bindDefault$1(EhCacheApi.scala:85):
> Binding(interface play.api.cache.SyncCacheApi to BindingKeyTarget(interface play.api.cache.SyncCacheApi qualified with QualifierInstance(@play.cache.NamedCache(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1).
>   at play.api.cache.redis.GuiceProvider$.defaultBinding$1(RedisCacheModule.scala:73):
> Binding(interface play.api.cache.SyncCacheApi to BindingKeyTarget(interface play.api.cache.SyncCacheApi qualified with QualifierInstance(@javax.inject.Named(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1)
> 
> 4) A binding to play.cache.AsyncCacheApi was already configured at play.api.cache.ehcache.EhCacheModule$$anonfun$$lessinit$greater$1.bindDefault$1(EhCacheApi.scala:85):
> Binding(interface play.cache.AsyncCacheApi to BindingKeyTarget(interface play.cache.AsyncCacheApi qualified with QualifierInstance(@play.cache.NamedCache(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1).
>   at play.api.cache.redis.GuiceProvider$.defaultBinding$1(RedisCacheModule.scala:73):
> Binding(interface play.cache.AsyncCacheApi to BindingKeyTarget(interface play.cache.AsyncCacheApi qualified with QualifierInstance(@javax.inject.Named(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1)
> 
> 5) A binding to play.cache.SyncCacheApi was already configured at play.api.cache.ehcache.EhCacheModule$$anonfun$$lessinit$greater$1.bindDefault$1(EhCacheApi.scala:85):
> Binding(interface play.cache.SyncCacheApi to BindingKeyTarget(interface play.cache.SyncCacheApi qualified with QualifierInstance(@play.cache.NamedCache(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1).
>   at play.api.cache.redis.GuiceProvider$.defaultBinding$1(RedisCacheModule.scala:73):
> Binding(interface play.cache.SyncCacheApi to BindingKeyTarget(interface play.cache.SyncCacheApi qualified with QualifierInstance(@javax.inject.Named(value=play)))) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1)
> 

1. Provide instructions to reproduce the issue
my build.sbt
```
import play.sbt.PlayImport

name := """svc"""

version := "1.0"

lazy val root = (project in file(".")).enablePlugins(PlayScala)

scalaVersion := "2.12.4"

libraryDependencies ++= Seq(
 "com.github.karelcemus" %% "play-redis" % "2.0.1",
"com.google.code.gson" % "gson" % "2.3.1",
  PlayImport.guice,
  PlayImport.cacheApi,
//  jdbc,
//  cache,
  ws,
  filters,
  openId
 )
fork in run := true
```

My controller code:
```
import play.api.cache.redis.CacheApi
@Singleton
class WxController @Inject()(cache:CacheApi) extends Controller 
```

My Application.conf
```
play.modules.disabled += "play.api.cache.EhCacheModule"
play.modules.enabled += "play.api.cache.redis.RedisCacheModule"
```

**Comment by KarelCemus** [comment]
Hi @YUFAN2GA,

that's strange. Looking at your `build.sbt` you do not include `ehcache` into your project. At least not directly, maybe transitively. So it should not fail in `EhCacheModule` in the first place.

Maybe you could double check that there is truly not imported the ehcache module. It shouldn't be.

Since Play 2.6 (and play-redis 2.x.x) it there exists `cacheApi` project, which does not include the ehcache thus you do not disable the module unless you import it (as you obviously do, somehow).

And since Play 2.6, the `EhCacheModule` was refactored into the different package, so to disable it, you have to disable `play.api.cache.ehcache.EhCacheModule`.

To sum it up:

1) you (probably transitively) include the `ehCache` module, which you shouldn't. I suggest to check it and to remove it if possible
2. otherwise change the fully classified name of the `EhCacheModule` to the correct updated name to disable the module and avoid the double binding.

Hope this helps, let me know.
